### PR TITLE
[JetNews] Integrate with new Snackbar + alpha02 bump

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt [hunk 1]
```diff
@@ -137,6 +139,24 @@ fun HomeScreen(
     navigateTo: (Screen) -> Unit,
     scaffoldState: ScaffoldState
 ) {
+
+    val errorMessage = stringResource(id = R.string.load_error)
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt:143]: Move this inside the if to avoid loading when not needed.

File: JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                favorites = emptySet(),
+                onToggleFavorite = {},
+                onRefreshPosts = {},
+                onErrorDismiss = {},
+                navigateTo = {},
+                scaffoldState = scaffoldState
+            )
+
+            val latch = CountDownLatch(1)
+            snapshotFlow { snackbarHostState.currentSnackbarData }
```
Comment (nickbutcher) [inline JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt:65]: Add a comment about `snapshotFlow`

File: JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                .onEach {
+                    if (it != null) {
```
Comment (nickbutcher) [inline JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt:67]: Comment explaining what this conditional means

File: JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @get:Rule
+    val composeTestRule = createComposeRule(disableTransitions = true)
+
+    @OptIn(
+        ExperimentalMaterialApi::class,
+        ExperimentalComposeApi::class
+    )
+    @Test
+    fun postsContainError_snackbarShown() {
+        composeTestRule.setContent {
```
Comment (nickbutcher) [inline JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt:50]: Add given/when/then comments to this test?

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt [hunk 5]
```diff
@@ -84,13 +81,14 @@ import kotlinx.coroutines.runBlocking
...
  */
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun HomeScreen(
     navigateTo: (Screen) -> Unit,
     postsRepository: PostsRepository,
     scaffoldState: ScaffoldState = rememberScaffoldState()
 ) {
-    val (postUiState, refreshPost, clearError) = launchUiStateProducer(postsRepository) {
+    val (postUiState, refreshPost, clearError: () -> Unit) = launchUiStateProducer(postsRepository) {
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt:91]: Was this explicit type added intentionally? Seems unbalanced only having it on one property.

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt [hunk 6]
```diff
@@ -122,11 +120,15 @@ fun HomeScreen(
  * Stateless composable is not coupled to any specific state management.
  *
  * @param posts (state) the data to show on the screen
+ * @param favorites (state) favorite posts
+ * @param onToggleFavorite (event) toggles favorite for a post
  * @param onRefreshPosts (event) request a refresh of posts
- * @param onErrorDismiss (event) request the current error be dismissed
+ * @param onErrorDismiss (event) indicate an error can be dismissed
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt:126]: Why the change? Isn't "request" correct here as the caller can ignore the event to veto it.

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt [hunk 7]
```diff
@@ -137,6 +139,22 @@ fun HomeScreen(
     navigateTo: (Screen) -> Unit,
     scaffoldState: ScaffoldState
 ) {
+    if (posts.hasError) {
+        val errorMessage = stringResource(id = R.string.load_error)
+        val retryMessage = stringResource(id = R.string.retry)
+        // Show snackbar message using a coroutine scope
+        launchInComposition(posts) {
+            val result = scaffoldState.snackbarHostState.showSnackbar(
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt:147]: `snackbarResult`?

**Comment by nickbutcher** [comment]
Thanks for the change. A few more comments, mostly requesting… comments.

File: JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt (hunk 3 above)
Comment (JoseAlcerreca) [reply JetNews/app/src/androidTest/java/com/example/jetnews/HomeScreenSnackbarTest.kt:67]: It's useless actually

File: JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt (hunk 5 above)
Comment (JoseAlcerreca) [reply JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreen.kt:91]: Left it accidentally. It's too much magic

**Review by nickbutcher — APPROVED** [review]
LGTM – Thanks for the changes. Looks like you have a stray import that needs spotless-ing.
